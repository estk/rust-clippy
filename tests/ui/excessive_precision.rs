#![feature(plugin, custom_attribute)]
#![warn(excessive_precision)]

fn main() {
    // TODO add prefix tests
    // Consts
    const GOOD32_SUF: f32 = 0.123_456_f32;
    const GOOD32: f32 = 0.123_456;
    const GOOD64: f64 = 0.123_456_789_012;

    const BAD32_SUF: f32 = 0.123_456_789_f32;
    const BAD32: f32 = 0.123_456_789;
    const BAD64: f64 = 0.123_456_789_012_345_67;
    const BAD64_SUF: f64 = 0.123_456_789_012_345_67f64;

    // const BAD64_32_SUF: f32 = 0.123_456_789_012_345_6f64;
    // Literal
    println!(9.999_999_999_999_999_999_999_999_999);

    // TODO add inferred type tests
    // Locals
    let good32_suf: f32 = 0.123_456_f32;
    let good32: f32 = 0.123_456;
    let good64: f64 = 0.123_456_789_012;

    let bad32_suf: f32 = 1.123_456_789_f32;
    let bad32: f32 = 1.123_456_789;
    let bad64: f64 = 0.123_456_789_012_345_67;
    let bad64_suf: f64 = 0.123_456_789_012_345_67f64;

    // TODO Vectors / nested vectors
    let recurse: Vec<f32> = vec![0.123_456_789];

    // Exponential float notation
    let good_exp32: f32 = 1e-10;
    let bad_exp32: f32 = 1.123456789e-10;
}
