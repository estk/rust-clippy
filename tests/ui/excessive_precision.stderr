error: float has excessive precision
  --> $DIR/excessive_precision.rs:14:28
   |
14 |     const BAD32_SUF: f32 = 0.123_456_789_f32;
   |                            ^^^^^^^^^^^^^^^^^ help: consider changing the type or truncating it to: `0.12345679`
   |
   = note: `-D excessive-precision` implied by `-D warnings`

error: float has excessive precision
  --> $DIR/excessive_precision.rs:15:24
   |
15 |     const BAD32: f32 = 0.123_456_789;
   |                        ^^^^^^^^^^^^^ help: consider changing the type or truncating it to: `0.12345679`

error: float has excessive precision
  --> $DIR/excessive_precision.rs:16:24
   |
16 |     const BAD64: f64 = 0.123_456_789_012_345_67;
   |                        ^^^^^^^^^^^^^^^^^^^^^^^^ help: consider changing the type or truncating it to: `0.12345678901234566`

error: float has excessive precision
  --> $DIR/excessive_precision.rs:17:28
   |
17 |     const BAD64_SUF: f64 = 0.123_456_789_012_345_67f64;
   |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: consider changing the type or truncating it to: `0.12345678901234566`

error: float has excessive precision
  --> $DIR/excessive_precision.rs:21:20
   |
21 |     println!("{}", 8.888_888_888_888_888_888_888);
   |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: consider changing the type or truncating it to: `8.88888888888889`

error: float has excessive precision
  --> $DIR/excessive_precision.rs:30:26
   |
30 |     let bad32_suf: f32 = 1.123_456_789_f32;
   |                          ^^^^^^^^^^^^^^^^^ help: consider changing the type or truncating it to: `1.1234568`

error: float has excessive precision
  --> $DIR/excessive_precision.rs:31:22
   |
31 |     let bad32: f32 = 1.123_456_789;
   |                      ^^^^^^^^^^^^^ help: consider changing the type or truncating it to: `1.1234568`

error: float has excessive precision
  --> $DIR/excessive_precision.rs:32:22
   |
32 |     let bad64: f64 = 0.123_456_789_012_345_67;
   |                      ^^^^^^^^^^^^^^^^^^^^^^^^ help: consider changing the type or truncating it to: `0.12345678901234566`

error: float has excessive precision
  --> $DIR/excessive_precision.rs:33:26
   |
33 |     let bad64_suf: f64 = 0.123_456_789_012_345_67f64;
   |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: consider changing the type or truncating it to: `0.12345678901234566`

error: float has excessive precision
  --> $DIR/excessive_precision.rs:36:34
   |
36 |     let recurse: Vec<f32> = vec![0.123_456_789];
   |                                  ^^^^^^^^^^^^^ help: consider changing the type or truncating it to: `0.12345679`

error: float has excessive precision
  --> $DIR/excessive_precision.rs:40:24
   |
40 |     let bad_e32: f32 = 1.123_456_788_888e-10;
   |                        ^^^^^^^^^^^^^^^^^^^^^ help: consider changing the type or truncating it to: `0.00000000011234568`

error: float has excessive precision
  --> $DIR/excessive_precision.rs:43:27
   |
43 |     let bad_bige32: f32 = 1.123_456_788_888E-10;
   |                           ^^^^^^^^^^^^^^^^^^^^^ help: consider changing the type or truncating it to: `0.00000000011234568`

error: aborting due to 12 previous errors

